chain/epoch-manager/src/lib.rs

```diff
@@ -1616,31 +1620,39 @@ impl EpochManager {
                                     endorsement_stats.expected;
                             }
                         }
-                        let mut shards = validator_to_shard[validator_id]
-                            .clone()
-                            .into_iter()
-                            .collect::<Vec<ShardId>>();
-                        shards.sort();
+                        // Collect the shards for which the validator was *expected* to produce at least one chunk.
+                        let mut shards_produced = chunks_stats_by_shard
+                            .iter()
+                            .filter_map(|(shard, stats)| (stats.expected() > 0).then_some(*shard))
+                            .collect_vec();
+                        shards_produced.sort();
+                        // Collect the shards for which the validator was *expected* to validate at least one chunk chunk.
```

Comment:
```suggestion
                        // Collect the shards for which the validator was *expected* to validate at least one chunk.
```

Reply:
shards_endorsed will always be all shards, because even if chunk validator has one mandate, in 43200 blocks it will likely be randomly selected for each shard at least one. But I also don't see a reason to define validator info differently.